### ISSUE TITLE
Login request to paywalled website doesn't seems to use the overriden user agent from siteconfig

### ISSUE THREAD

**Issue by apiontek** [wallabag/wallabag#3654]
### Issue details

I was setting up a wallabag site config for myself for dissentmagazine.org, and it's working aside from the login with site credentials.

As best I can tell, the site config info should be:
```
login_uri: https://www.dissentmagazine.org/login
login_username_field: username
login_password_field: password
```

But I get the following result, in wallabag:
```
Your IP: aaa.bbb.ccc.ddd
URL: www.dissentmagazine.org/login
Your Browser: Guzzle/5.3.1 curl/7.58.0 PHP/7.2.5-0ubuntu0.18.04.1
Block ID: DDOS22
Block reason: DDOS attempt was blocked.
Time: Fri May 25 16:07:30 2018
Server ID: 14004
```

Is there any way around this sort of thing?
### Environment

* wallabag version: 2.3.2 (git HEAD at tag 2.3.2)
* How did you install wallabag? git clone
* php version: 7.2.5
* OS: Ubuntu 18.04
* type of hosting (shared or dedicated): home server
* storage system: MariaDB 10.1.29

### Steps to reproduce/test case

create site config with login info (mine quoted below), try to access a locked article such as [this one](https://www.dissentmagazine.org/article/ideology-fossil-fuels-apocalypse-petrocapitalism-energy-humanities)

```
tidy: yes
prune: yes

## title
title: //h1[contains(@class, 'hide-on-small-only')]
title: //h2[contains(@class, 'hide-on-med-and-up')]

## author
author: //aside[@class='article-meta']/*[contains(@class, 'author')]

## date
strip: //aside[@class='article-meta']/img[@class='emoji']
replace_string(&squarf;): <br />
date: //aside[@class='article-meta']/text()
date: //aside[@class='article-meta']/a[contains(@href, '/issue/')]

## body
strip: //aside[@class='article-meta']
strip: //figure[@class='article-image']/img
strip: //div[@class='excerpt-tag']
body: //*[@id="article"]/div/article

requires_login: yes

login_uri: https://www.dissentmagazine.org/login
login_username_field: username
login_password_field: password

not_logged_in_xpath: //*[@id="article"]/div/article/div[@class="login"]
```

**Comment by j0k3r** [comment]
I guess the problem come from: `Your Browser: Guzzle/5.3.1 curl/7.58.0 PHP/7.2.5-0ubuntu0.18.04.1
`

You should override the User-Agent value. Maybe it'll fix the problem.
> http_header(user-agent): Mozilla/5.0 (iPhone; CPU iPhone OS 10_2 like Mac OS X) AppleWebKit/602.1.32 (KHTML, like Gecko) Mobile/14C92

**Comment by apiontek** [comment]
How would one go about overriding the user-agent value for this functionality? Is there someplace in the wallabag interface to set that, or can you point me to the code to hack at?

**Comment by apiontek** [comment]
This doesn't work. It doesn't seem to change the user-agent string at all.

I put the following at the top of my `dissentmagazie.org.txt` file:

`http_header(user-agent): Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/51.0.2704.103 Safari/537.36`

I forced re-fetch on an article I'd added before, and tried adding a new paywalled article -- both failed, and the error response that gets fetched instead, still shows "Guzzle/5.3.1 curl/7.58.0 PHP/7.2.5-0ubuntu0.18.04.1" as the user agent.

However, if I coment out the lines related to site credentials (requires_login, login_uri, login_username_field, login_password_field, not_logged_in_xpath) and just fetch a regular, non-paywalled article, everything works fine.

So something about the mechanism that handles the site credentials is sending that "Guzzler" user agent, and putting the "http_header(user-agent)" line in the site-config is not altering that user agent.

**Comment by j0k3r** [comment]
Yeah maybe the login request doesn't get the overriden user agent and use the default one.

**Comment by apiontek** [comment]
OK, I know this is old now, but I decided to investigate this again, with a little more experience under my belt...

The login request doesn't get the overridden user-agent - but I can force it by modifying `src/Wallabag/CoreBundle/Helper/HttpClientFactory.php` to:

```php
$client = new Client(['handler' => new SafeCurlHandler(), 'defaults' => [
            'cookies' => $this->cookieJar,
            'headers' => [
                'User-Agent' => 'Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:15.0) Gecko/20100101 Firefox/15.0.1'
            ]
        ]]);
```

Then the "site credentials" login request uses the modified User-Agent -- but the login request still results in the same DDOS protection error, just showing the modified User-Agent string.

But if I mock up my own quick GuzzleHttp script to do the same POST request, or do the request in Postman, or with curl -- it works. I get a cookie and a full page loads, no DDOS.

My quick Guzzle 5 script:

```php
<?php
require 'vendor/autoload.php';
use GuzzleHttp\Client;
$cookieJar = new GuzzleHttp\Cookie\CookieJar();
$client = new Client(['defaults' => [
  'cookies' => $cookieJar,
  'headers' => [
      'User-Agent' => 'Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:15.0) Gecko/20100101 Firefox/15.0.1'
  ]
]]);
$client->setDefaultOption('verify', false);
$response = $client->get('https://www.dissentmagazine.org/login?username=MY@EMAIL.COM&password=MYPASS');

print($response);
print_r($cookieJar->toArray());
?>
```

I tried enabling debug logs in my wallabag instance and re-fetching the content but no logs were output so I'm not sure if I did that wrong:

```yaml
monolog:
    handlers:
        main:
            type: fingers_crossed
            action_level: error
            handler: nested
            channels: ['!graby']
        nested:
            type: stream
            path: "/var/www/my.domain/wallabag/logs/prod.log"
#            path: "%kernel.logs_dir%/%kernel.environment%.log"
            level: debug
        graby:
            type: stream
            path: "/var/www/my.domain/wallabag/logs/graby.log"
#            path: "%kernel.logs_dir%/graby.log"
            level: info
            channels: ['graby']
        console:
            type: console
```

I changed the log paths in case there was a write-protection issue for the account. But this wouldn't catch Guzzle logs, would it? How can I enable debug logging for Guzzle?

I'm happy to privately provide my account credentials & current site config for this site to to get this resolved.

**Comment by apiontek** [comment]
Just adding that currently I'm on:

Wallabag 2.3.8
PHP 7.2.19
Ubuntu 18.04.2 LTS
Postgresql 10.9

Installed per [instructions](https://doc.wallabag.org/en/admin/installation/installation.html) with git clone and make install -- and kept up-to-date roughly weekly with `su -s /bin/bash -c "make update" www-data`

**Comment by j0k3r** [comment]
The main problem is that the "plugin" which handle login to website isn't as much as I want integrated into Graby. That's why you don't have the defined user agent.
I would like to integrate that plugin into the core of Graby in a better way.
